src/util/pool/poolUtils.ts

```diff
@@ -9,41 +9,55 @@ import {
   encodeString
 } from "../util";
 import {AccountInformation} from "../account/accountTypes";
-import {tinymanContract} from "../../contract/contract";
+import {
+  ContractVersionValue,
+  CONTRACT_VERSION,
+  tinymanContract_v1_1,
+  tinymanContract_v2
+} from "../../contract/contract";
 import {PoolInfo, PoolReserves, PoolStatus} from "./poolTypes";
+import {SupportedNetwork} from "../commonTypes";
+import {getValidatorAppID} from "../../validator";
+import {getIsV2ContractVersion} from "../../contract/utils";
 
 /**
  * Look up information about an pool.
  *
- * @param client An Algodv2 client.
- * @param pool Parameters of the pool to look up.
- * @param pool.validatorAppID The ID of the Validator App for the network.
- * @param pool.asset1ID The ID of the first asset in the pool pair.
- * @param pool.asset2ID The ID of the second asset in the pool pair.
+ * @param params - The parameters for the pool information request.
+ * @param {Algodv2} params.client An Algodv2 client.
+ * @param {SupportedNetwork} params.network Network to use.
+ * @param {ContractVersion} params.contractVersion Contract version to use.
+ * @param {number} params.asset1ID The ID of the first asset in the pool pair.
+ * @param {number} params.asset2ID The ID of the second asset in the pool pair.
  */
-export async function getPoolInfo(
-  client: any,
-  pool: {
-    validatorAppID: number;
-    asset1ID: number;
-    asset2ID: number;
-  }
-): Promise<PoolInfo> {
-  const poolLogicSig = tinymanContract.getPoolLogicSig(pool);
+export async function getPoolInfo(params: {
+  client: Algodv2;
+  network: SupportedNetwork;
+  contractVersion: ContractVersionValue;
+  asset1ID: number;
+  asset2ID: number;
+}): Promise<PoolInfo> {
+  const {client, network, contractVersion, asset1ID, asset2ID} = params;
+  const contract =
+    contractVersion === CONTRACT_VERSION.V1_1 ? tinymanContract_v1_1 : tinymanContract_v2;
```

Comment:
We repeat this logic at severeal places. Can we create a utility and re-use it whenevery necessary?
```suggestion
  const contract = getContract(contractVersion); // or `getContractInstance`
```
This is also better for maintainability. If we need to add v3, we just need to change logic in `getContract`